source/FreeRTOS_IPv6.c

```diff
@@ -173,7 +173,21 @@ const struct xIPv6_Address FreeRTOS_in6addr_loopback = { { 0, 0, 0, 0, 0, 0, 0,
             }
             else if( ucNextHeader == ( uint8_t ) ipPROTOCOL_ICMP_IPv6 )
             {
-                uxMinimumLength = ipSIZE_OF_ETH_HEADER + ipSIZE_OF_IPv6_HEADER + uxExtHeaderLength + ipSIZE_OF_ICMPv6_HEADER;
+                uint8_t ucTypeOfMessage;
+
+                uxMinimumLength = ipSIZE_OF_ETH_HEADER + ipSIZE_OF_IPv6_HEADER + uxExtHeaderLength;
+
+                ucTypeOfMessage = pucEthernetBuffer[ uxMinimumLength ];
+
+                if( ( ucTypeOfMessage == ipICMP_PING_REQUEST_IPv6 ) ||
+                    ( ucTypeOfMessage == ipICMP_PING_REPLY_IPv6 ) )
+                {
+                    uxMinimumLength += sizeof( ICMPEcho_IPv6_t );
+                }
+                else
+                {
+                    uxMinimumLength += ipSIZE_OF_ICMPv6_HEADER;
```

Comment:
From [ICMPv6](https://en.wikipedia.org/wiki/ICMPv6):
- RA header might have only 16 bytes.
- RS header might have only 8 bytes. (But we don't handle this now, it's fine to skip this.)

ipSIZE_OF_ICMPv6_HEADER is 24 now, is it ok for RA packets?

Reply:
Although my IPv6 tests didn't show any warnings, I will add two checks:

~~~
else if( ucTypeOfMessage == ipICMP_ROUTER_SOLICITATION_IPv6 )
{
    uxMinimumLength += sizeof( ICMPRouterSolicitation_IPv6_t );
}
else if( ucTypeOfMessage == ipICMP_ROUTER_ADVERTISEMENT_IPv6 )
{
    uxMinimumLength += sizeof( ICMPRouterAdvertisement_IPv6_t );
}
~~~

Thank you.